### DISCUSSION TITLE
Could use some help with my GUI based java program

### DISCUSSION THREAD

**Discussion by Mstu0046** [community/community#114791] [Programming Help]
### Body

I began creating a turn based fighting sim after learning GUI. I ran into some issues when attempting to update my JLabel as the program was running. It calculates and holds the proper values but the Enemyhp display is never corrected after the user attacks. I'm not sure what I could do to sort the issue as I've tried a many things.

Here is the "Enemy" class
package combatsim;

import javax.swing.*;

/**
   The PlayerSPanel class displays the player stats in a panel.
*/

public class EnemySPanel extends JPanel
{
    //creates fields for enemy stats
    private double hp = 200;
    private double dmg = 10;
    //creates labels to hold enemy stats
    private JLabel enemy;
    private JLabel health;
    private JLabel damage;
   /**
      Constructor
   */
   public EnemySPanel()
   {
      // Create the label.
      enemy = new JLabel("Enemy Stats");
      health = new JLabel("");
      damage = new JLabel("");
      //sets labels to the field values
      health.setText(Double.toString(hp));
      damage.setText(Double.toString(dmg));

      // Add the label to this panel.
      add(enemy);
      add(health);
      add(damage);
   }
   
   public double GetHealth()
   {   
       return hp;
   }
   
   public double GetDamage()
   {   
       return dmg;
   }
   
   public void SetHealth(String eh)
   {
       String EHealth = eh;
       
       health.setText(EHealth);
   }
   
   public void SetDamage(String ed)
   {   
       String EDamage = ed;
       
       damage.setText(EDamage);
   }
   
}

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ethanwater** [comment]
You're not even applying the changes to the enemy object in your SetDamage script. You should probably do that!
